pocs/utils/images/fits.py

```diff
@@ -320,3 +320,50 @@ def fpack(fits_fname, unpack=False, verbose=False):
         output, errs = proc.communicate()
 
     return out_file
+
+
+def write_fits(data, header, filename, logger, exposure_event=None):
+    """
+    Write FITS file to requested location
+    """
+    hdu = fits.PrimaryHDU(data, header=header)
+
+    # Create directories if required.
+    if os.path.dirname(filename):
+        os.makedirs(os.path.dirname(filename), mode=0o775, exist_ok=True)
+
+    try:
+        hdu.writeto(filename)
+    except OSError as err:
+        logger.error('Error writing image to {}!'.format(filename))
+        logger.error(err)
+    else:
+        logger.debug('Image written to {}'.format(filename))
+    finally:
+        # Set event to mark exposure complete.
+        if exposure_event is not None:
+            exposure_event.set()
+
+
+def update_headers(filepath, info):
+    with fits.open(file_path, 'update') as f:
+        hdu = f[0]
+        hdu.header.set('IMAGEID', info.get('image_id', ''))
+        hdu.header.set('SEQID', info.get('sequence_id', ''))
+        hdu.header.set('FIELD', info.get('field_name', ''))
+        hdu.header.set('RA-MNT', info.get('ra_mnt', ''), 'Degrees')
+        hdu.header.set('HA-MNT', info.get('ha_mnt', ''), 'Degrees')
+        hdu.header.set('DEC-MNT', info.get('dec_mnt', ''), 'Degrees')
+        hdu.header.set('EQUINOX', info.get('equinox', 2000.))  # Assume J2000
```

Comment:
I think this changed again so we just need to make sure it's okay in the end.

Reply:
I took the latest version from `cr2_to_fits` but yes, would be good to test.